File: app/src/main/java/ihsanbal/com/logginginterceptor/di/NetComponent.java
package ihsanbal.com.logginginterceptor.di;

import javax.inject.Singleton;

import dagger.Component;
import ihsanbal.com.logginginterceptor.di.NetModule;
import ihsanbal.com.logginginterceptor.ui.MainActivity;

/**
 * @author ihsan on 09/02/2017.
 */

@Singleton
@Component(modules = {NetModule.class})
public interface NetComponent {
    void inject(MainActivity activity);
}

File: app/src/main/java/ihsanbal/com/logginginterceptor/di/NetComponent.kt
package ihsanbal.com.logginginterceptor.di

import dagger.Component
import ihsanbal.com.logginginterceptor.ui.MainActivity
import javax.inject.Singleton

/**
 * @author ihsan on 09/02/2017.
 */
@Singleton
@Component(modules = [NetModule::class])
interface NetComponent {
    fun inject(activity: MainActivity?)
}